qiskit_ibm_runtime/sampler.py

```diff
@@ -288,10 +286,7 @@ def _call(
             options=Options._get_runtime_options(combined),
         ).result()
 
-        return SamplerResult(
-            quasi_dists=raw_result["quasi_dists"],
-            metadata=raw_result["metadata"],
-        )
+        return raw_result
```

Comment:
```suggestion
        return SamplerResult(
            quasi_dists=raw_result["quasi_dists"],
            metadata=raw_result["metadata"],
        )
```
This should return `SamplerResult` since we no longer pass `SamplerResultDecoder` to the `run()` above (to make the quasi dist change non-breaking)

Reply:
The `raw_result` is already a `SamplerResult` because if we want to always return the correct types I added the following to `result()`. So I think it would have to be a breaking change 

```python
if self.program_id == "sampler":
      self._result_decoder = SamplerResultDecoder
elif self.program_id == "estimator":
      self._result_decoder = EstimatorResultDecoder
```

Reply:
We _could_ pass in another decoder that doesn't convert `quasi_dists` to `QuasiDistribution`. But it's probably not worth the effort, and `QuasiDistribution` is the correct type after all.